dqsegdb2/http.py

```diff
@@ -43,12 +54,57 @@ def request(url, **urlopen_kw):
     reponse : `http.client.HTTPResponse`
         the reponse from the URL
     """
+    req = Request(url)
     if urlparse(url).scheme == 'https' and 'context' not in urlopen_kw:
         from ssl import create_default_context
-        from gwdatafind.utils import find_credential
         urlopen_kw['context'] = context = create_default_context()
-        context.load_cert_chain(*find_credential())
-    return urlopen(url, **urlopen_kw)
+        aud = urlparse(url).hostname
+        token_data = None
+        tok = None
+
+        # look for a condor-created scitoken for the target url
+        if os.environ.get('_CONDOR_CREDS'):
+            for f in os.listdir(os.environ['_CONDOR_CREDS']):
+                if f.endswith(".use"):
+                    try:
+                        with open(os.path.join(
+                                  os.environ['_CONDOR_CREDS'],f)) as t:
+                            token_data = t.read().strip()
+                            tok = scitokens.SciToken.deserialize(token_data,
+                                                                 audience=aud)
+                            break
+                    except (InvalidTokenFormat, InvalidAudienceError):
+                        # ignore files that do not contain tokens
+                        # or that are for a different audience
+                        pass
+
+        # use the token in the specified file
+        if os.environ.get('SCITOKENS_FILE'):
+            with open(os.environ.get('SCITOKENS_FILE')) as t:
+                token_data = t.read().strip()
+                tok = scitokens.SciToken.deserialize(token_data,
+                                                     audience=aud)
+
+        # use a serialized token in an environment variable
+        if os.environ.get('SCITOKEN'):
+            token_data = os.environ['SCITOKEN'].strip()
+            tok = scitokens.SciToken.deserialize(token_data,
+                                                 audience=aud)
```

Comment:
Can you please modularise this into a separate function, mainly to make it easier to maintain?